src/main/java/edu/umich/tl/ApiCallHandler.java

```diff
@@ -20,44 +20,69 @@
 	private String canvasURL=CourseDelete.canvasURL;
 	private String esbURL=CourseDelete.esbURL;
 	private CanvasCallEnum canvasCall;
-	
 	private static String canvasToken=CourseDelete.canvasToken;
 	private static final String PER_PAGE = "per_page=100";
 	private static final String API_VERSION = "/api/v1";
 	
-	public ApiCallHandler(CanvasCallEnum apiCallType) {
-		this.canvasCall=apiCallType;
+	public ApiCallHandler(CanvasCallEnum canvasCall) {
+		this.canvasCall=canvasCall;
 	}
 	
-	public enum CanvasApiEnum{
-		TERM, UNPUBLISHED_COURSE_LIST;
+	public enum RequestTypeEnum{
+		TERM, UNPUBLISHED_COURSE_LIST, UNPUBLISHED_COURSE_LIST_PAGINATION_URL;
 	}
 
-	public HttpResponse getApiResponse(CanvasApiEnum canvasApi) {
+	public HttpResponse getApiResponse(RequestTypeEnum requestType, String canvasTermIdForSisTermId, String url) {
 		HttpResponse httpResponse = null;
-		String url = null;
-		switch (canvasApi) {
+		String urlSuffix;
+		switch (requestType) {
 		case TERM:
-			String urlSuffix=API_VERSION+"/accounts/1/terms?"+PER_PAGE;
-			if (canvasCall.equals(CanvasCallEnum.API_DIRECT_CANVAS)) {
-				url = canvasURL + urlSuffix;
-				httpResponse = apiDirectCanvas(url);
-			} else if (canvasCall.equals(CanvasCallEnum.API_ESB_CANVAS)) {
-				url = esbURL + urlSuffix;
-				httpResponse =apiESBCanvas(url);
-			}
-			M_log.debug("TermURL: " + url);
+			urlSuffix=API_VERSION+"/accounts/1/terms?"+PER_PAGE;
+			httpResponse = apiCallToCanvas(urlSuffix);
 			break;
 		case UNPUBLISHED_COURSE_LIST:
+			urlSuffix=API_VERSION+"/accounts/1/courses?enrollment_term_id="+canvasTermIdForSisTermId+"&published=false&"+PER_PAGE;
+			httpResponse = apiCallToCanvas(urlSuffix);
+			break;
+			// we are having separate case for unpublished course list since pagination object in the response header has fully framed url and hence we can use it directly,
+			// we still need to make the check if this call is to 'direct' or 'esb' canvas. We can get info from the 'CanvasCallEnum'
+		case UNPUBLISHED_COURSE_LIST_PAGINATION_URL:
+			if (isThisIsADirectCanvasCall()) {
+				httpResponse=apiDirectCanvas(url);
+			}else if (isThisAEsbCanvasCall()) {
+				httpResponse=apiESBCanvas(url);
+			}
 			break;
 		default:
 			break;
 		}
 		return httpResponse;
 	}
-
 	
+	private HttpResponse apiCallToCanvas(String urlSuffix) {
+		String url = null;
+		HttpResponse httpResponse = null;
+		if (isThisIsADirectCanvasCall()) {
+			url = canvasURL + urlSuffix;
+			httpResponse = apiDirectCanvas(url);
+		} else if (isThisAEsbCanvasCall()) {
+			url = esbURL + urlSuffix;
+			httpResponse =apiESBCanvas(url);
+		}
+		M_log.debug("The Api call URL: " + url);
```

Comment:
Would it be easy to add the HTTP status code from the response? That would provide a clue in the log if something is going wrong.